test/crawler/keyword/keywords_test.exs

```diff
@@ -5,15 +5,33 @@ defmodule Crawler.Keyword.KeywordsTest do
   alias Crawler.Keyword.Schemas.Keyword
   alias CrawlerWorker.Keyword.CrawlerWorker
 
-  describe "list_keywords/1" do
+  describe "list_keywords_by_filter_params/1" do
     test "given a valid user id, returns user's keyword list" do
       %{id: user_id} = insert(:user)
       insert(:keyword, user_id: user_id)
 
       %{id: expected_user_id} = insert(:user)
       expected_keyword = insert(:keyword, user_id: expected_user_id)
 
-      assert Keywords.list_keywords(expected_user_id) == [expected_keyword]
+      assert Keywords.list_keywords_by_filter_params(expected_user_id) == [expected_keyword]
+    end
+  end
+
+  describe "list_keywords_by_filter_params/2" do
+    test "given a valid user id and valid keyword name, returns user's keyword list" do
+      %{id: user_id} = insert(:user)
+      insert(:keyword, user_id: user_id, name: "keyword")
+
+      %{id: expected_user_id} = insert(:user)
+      expected_keyword = insert(:keyword, user_id: expected_user_id, name: "phone")
+      insert(:keyword, user_id: expected_user_id, name: "tv")
+
+      assert [%{name: stored_keyword_name}] =
+               Keywords.list_keywords_by_filter_params(expected_user_id, %{
+                 "keyword" => expected_keyword.name
```

Comment:
should we test for the invalid keyword 🤔

Reply:
also should we add a system test for this feature @vnntsu

Reply:
and we can add test for partial search, for example `hon` to return `phone` keyword

Reply:
Yes, I added these cases in [7f408e0](https://github.com/vnntsu/elixir-ic/pull/44/commits/7f408e06e4106fc62f359133bf1b04b38e290b2c)
Added feature test as well.

Thanks for pointing them out. 🙏